packages/web-pkg/src/components/Search/types.ts

```diff
@@ -1,8 +1,16 @@
 import { SearchResource } from '@ownclouders/web-client/src/webdav/search'
+import { Component } from 'vue'
+
+export interface GenericSearchResultItem {
+  id: string
+  name: string
+  icon?: string
+}
 
 export interface SearchResultValue {
-  id: string | number
-  data: SearchResource
+  id: string
+  // FIXME: SearchResource is very specific for webdav search
+  data: GenericSearchResultItem | SearchResource
```

Comment:
Just thinking, do we really need the `SearchResource` here then? 🤔 Shouldn't it be as generic as possible in `web-pkg`?

Search implementations like the webdav search could still have their own type, as long as they implement the generic type.

Reply:
The purpose of this PR was to widen the current typing a little, I don't have a strong opinion on how generic it ideally _should_ be 😇

Reply:
Alright, I guess the purpose of this PR is served with this 😄